File: processes/order/src/main/kotlin/io/miragon/zeebe/tm/order/adapter/in/rest/model/FormDto.kt
package io.miragon.zeebe.tm.order.adapter.`in`.rest.model

import io.miragon.zeebe.tm.order.adapter.`in`.rest.model.schema.FormDataDto

sealed class FormDto
{
    data class JsonForm<T : FormDataDto>(
        val schema: String,
        val uiSchema: String,
        val updatable: Boolean,
        val formData: T?
    ) : FormDto()

    data class HtmlForm<T : FormDataDto>(
        val html: String,
        val updatable: Boolean,
        val formData: T?
    ) : FormDto()
}

File: apps/tasklist/src/client/process/models/FormDto.ts
export type FormDto = HtmlFormDto | JsonFormDto;

export interface HtmlFormDto {
    html: string;
    updatable: boolean;
    formData?: any;
}

export interface JsonFormDto {
    schema: string;
    uiSchema: string;
    updatable: boolean;
    formData?: any;
}
